src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -35,19 +35,31 @@ internal class FileConfig
         {
             private static readonly Regex AdditionalOptionsSplitPattern = new Regex("(?<=^[^\"]*(?:\"[^\"]*\"[^\"]*)*) (?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
 
-            public static FileConfig TryGet(ProjectItem projectItem, string absoluteFilePath)
+            public static FileConfig TryGet(ProjectItem dteProjectItem, string absoluteFilePath)
             {
-                string configurationName = projectItem.ConfigurationManager.ActiveConfiguration.ConfigurationName; // "Debug" or "Release"
-                string platformName = projectItem.ConfigurationManager.ActiveConfiguration.PlatformName; // "Win32" or "x64"
-                string pattern = configurationName + '|' + platformName;
+                var vcProject = dteProjectItem.ContainingProject.Object as VCProject;
+                var file = dteProjectItem.Object as VCFile;
+                if (vcProject == null || file == null)
+                {
+                    return null;
+                }
 
-                dynamic project = projectItem.ContainingProject.Object; // Microsoft.VisualStudio.VCProjectEngine.VCProject
-                dynamic config = project.Configurations.Item(pattern); // Microsoft.VisualStudio.VCProjectEngine.VCConfiguration
+                var vcConfig = vcProject.ActiveConfiguration;
```

Comment:
As discussed offline, this fixes #1537 - we're using the API to get the active configuration rather than trying to fetch it based on a name that we calculate from other properties.